mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java

```diff
@@ -130,16 +134,24 @@ private void subscribeWorkQueue() {
             .map(eventInBytes -> new String(eventInBytes, StandardCharsets.UTF_8))
             .map(eventSerializer::fromJson)
             .map(JsResult::get)
+            .flatMap(event -> deliverEvent(mailboxListener, event))
             .subscribeOn(Schedulers.elastic())
-            .subscribe(event -> deliverEvent(mailboxListener, event)));
+            .subscribe());
     }
 
-    private void deliverEvent(MailboxListener mailboxListener, Event event) {
-        try {
-            mailboxListener.event(event);
-        } catch (Exception e) {
-            LOGGER.error("Exception happens when handling event of user {}", event.getUser().asString(), e);
-        }
+    private Mono<Void> deliverEvent(MailboxListener mailboxListener, Event event) {
```

Comment:
That is not the original idea for RabbitMQ eventBus handling...

The design is to push the retry down to rabbitMQ in order to perform retries on other nodes (a specific node can be failing for specific reasons : code, network partitions, etc...)

We do acknoledge the delivery once done or reject it.